packages/playground/blueprints/src/lib/steps/activate-plugin.ts

```diff
@@ -65,22 +63,102 @@ export const activatePlugin: StepHandler<ActivatePluginStep> = async (
+			DOCROOT: docroot,
+		},
 	});
-	if (result.text !== 'Plugin activated successfully') {
-		logger.debug(result);
-		throw new Error(
-			`Plugin ${pluginPath} could not be activated – WordPress exited with no error. ` +
-				`Sometimes, when $_SERVER or site options are not configured correctly, ` +
-				`WordPress exits early with a 301 redirect. ` +
-				`Inspect the "debug" logs in the console for more details`
+	if (activatePluginResult.text) {
+		logger.warn(
+			`Plugin ${pluginPath} activation printed the following bytes: ${activatePluginResult.text}`
 		);
 	}
+
+	/**
+	 * Instead of checking the plugin activation response,
+	 * check if the plugin is active by looking at the active plugins list.
+	 *
+	 * We have to split the activation and the check into two PHP runs
+	 * because some plugins might redirect during activation,
+	 * which would prevent any output that happens after activation from being returned.
+	 *
+	 * Relying on the plugin activation response is not reliable because if the plugin activation
+	 * produces any output, WordPress will assume it's an activation error.
+	 * See WordPress source code for more details:
+	 * https://github.com/WordPress/wordpress-develop/blob/6.7/src/wp-admin/includes/plugin.php#L733
+	 *
+	 * Because some plugins can create an output, we need to use output buffering
+	 * to ensure the 'true' response is not polluted by other outputs.
+	 * If the plugin activation fails, we will return the buffered output as it might
+	 * contain more information about the failure.
+	 */
+	const isActiveCheckResult = await playground.run({
+		code: `<?php
+			ob_start();
+			require_once( getenv( 'DOCROOT' ) . "/wp-load.php" );
+
+			/**
+			 * Extracts the relative plugin path from either an absolute or relative plugin path.
+			 *
+			 * Absolute paths starting with plugin directory (e.g., '/wordpress/wp-content/plugins/test-plugin/index.php')
+			 * should be converted to relative paths (e.g., 'test-plugin/index.php')
+			 *
+			 * Directories should finish with a trailing slash to ensure we match the full plugin directory name.
+			 *
+			 * Examples:
+			 * - '/wordpress/wp-content/plugins/test-plugin/index.php' → 'test-plugin/index.php'
+			 * - '/wordpress/wp-content/plugins/test-plugin/' → 'test-plugin/'
+			 * - '/wordpress/wp-content/plugins/test-plugin' → 'test-plugin/'
+			 * - 'test-plugin/index.php' → 'test-plugin/index.php'
+			 * - 'test-plugin/' → 'test-plugin/'
+			 * - 'test-plugin' → 'test-plugin/'
+			 */
+			$plugin_directory = getenv( 'DOCROOT' ) . "/wp-content/plugins/";
+			$relative_plugin_path = getenv( 'PLUGIN_PATH' );
+			if (strpos($relative_plugin_path, $plugin_directory) === 0) {
+				$relative_plugin_path = substr($relative_plugin_path, strlen($plugin_directory));
+			}
+			if (
+				pathinfo($relative_plugin_path, PATHINFO_EXTENSION) !== 'php' &&
+				substr($relative_plugin_path, -1) !== '/'
+			) {
+				$relative_plugin_path = $relative_plugin_path . '/';
```

Comment:
nice!

Reply:
I'd add an `is_dir()` check

Reply:
> I'd add an is_dir() check

Good call, I added it in [dde935a](https://github.com/WordPress/wordpress-playground/pull/2066/commits/dde935a546bd03e1a6bf8b0838e93f6e6e42b365).

Reply:
Or, for brevity:

```php
			if ( is_dir( $plugin_directory . $relative_plugin_path ) ) {
				$relative_plugin_path = rtrim( $relative_plugin_path, '/' ) . '/';
			}
```

Reply:
Thanks, this is much cleaner now. bb24b49757cde44936c9307a84aef3bb3873faf1